File: python/python-code/day01/day01.py
# Check Data types
print(type([1, 2, 3]))          # list
print(type({'name': 'pacman'})) # dict
print(type((2, 4, 5)))          # tuple
print(type({4.5, 9}))           # set
print(type(1 + 1j))             # complex
print(type(True))               # bool
print(type(zip([1,2],[3,4])))   # set
print(3 ** 2)
print(3 // 2)

# Casting
num_int = 10
print('num_int',num_int)         # 10
num_float = float(num_int)
print('num_float:', num_float)   # 10.0

# float to int
gravity = 9.81
print(int(gravity))             # 9

# int to str
num_int = 10
print(num_int)                  # 10
num_str = str(num_int)
print(num_str)                  # '10'

# str to int
num_str = '10.6'
print('num_int', int(num_str))      # 10
print('num_float', float(num_str))  # 10.6

# str to list
first = 'Asabeneh'
print(first_name)
print(first_name)                    # 'Asabeneh'
first_name_to_list = list(first_name)
print(first_name_to_list)            # ['A', 's', 'a', 'b', 'e', 'n', 'e', 'h']

File: javascript/30_days_of_js/day01/day01.js
let firstName;
console.log(firstName); // not defined, because it is not assigned to a value yet
let emptyValue = null   // null in JavaScript means an empty value.
console.log(typeof 'Asabeneh') // string
console.log(typeof 5)          // number
console.log(typeof true )      // boolean
console.log(typeof null)       // object type
console.log(typeof undefined)  // undefined

// Declaring different variables of different data types
firstName = 'Asabeneh'  // first name of a person
let lastName = 'Yetayeh'    // last name of a person
let country = 'Finland'     // country
let city = 'Helsinki'       // capital city
let age = 100               // age in years
let isMarried = true

console.log(firstName, lastName, country, city, age, isMarried); //Asabeneh, Yetayeh, Finland, Helsinki, 100, True

// Declaring variables with number values
const gravity = 9.81      // earth gravity  in m/s2
const boilingPoint = 100  // water boiling point, temperature in oC
const PI = 3.14           // geometrical constant

console.log(gravity, boilingPoint, PI); // 9.81, 100, 3.14
// Variables can also be declaring in one line separated by comma
let name = 'Asabeneh',
  job = 'teacher',
  live = 'Finland';
console.log('my name is ' + name, "and I am a " + job, "in " + live);